spec/integration/bisect_spec.rb

```diff
@@ -33,14 +33,19 @@ def bisect(cli_args, expected_status=nil)
       end
     end
 
-    context "when the bisect commasaturingnd is long" do
+    context "when the bisect command saturates the pipe" do
       # On OSX and Linux a file descriptor limit meant that the bisect process got stuck at a certain limit.
       # This test demonstrates that we can run large bisects above this limit (found to be at time of commit).
       # See: https://github.com/rspec/rspec-core/pull/2669
       it 'does not hit pipe size limit and does not get stuck' do
         output = bisect(%W[spec/rspec/core/resources/blocking_pipe_bisect_spec.rb_], 1)
         expect(output).to include("No failures found.")
       end
+
+      it 'does not leave zombie processes', :unless => RSpec::Support::OS.windows? do
+        bisect(%W[spec/rspec/core/resources/blocking_pipe_bisect_spec.rb_], 1)
+        expect(%x[ps -ho pid,state]).to_not include("Z")
```

Comment:
Its worth noting this will fail if you have *any* zombies, if we know the pids of the bisector it can be targetted.

Reply:
I think it's good enough to start.

Reply:
The problem I have with this spec is we do not check Zombie process while bisect command is running. 

rspec-core before this patch:

![bisect_nothing](https://user-images.githubusercontent.com/8417720/84673830-7c78cc00-af2a-11ea-9b2a-f3c8ba4ea6f4.gif)

`rspec-core` with this patch with `detach`:

![bisect_detach](https://user-images.githubusercontent.com/8417720/84673863-84d10700-af2a-11ea-8e23-28f03a4d1fa4.gif)

I use this ugly code for this demo:

```ruby
require 'spec_helper'

RSpec.describe "bisect scenario for wrong spec isolation" do
  $global_bool = true

  describe 'mutate global var' do
    it do
      sleep 1
      $global_bool = false
      expect($global_bool).to eq true
    end
  end

  describe 'side effect' do
    it do
      sleep 1
      expect($global_bool).to eq true
    end
  end
end
```

We have to monitor the execution of bisect and inspect log after.

Reply:
Never mind it's great with your proposal. I thought the zombies were killed already when `bisect()` has done but it seems there is a slight delay.

🙏